sdl_android/src/main/java/com/smartdevicelink/api/lockscreen/LockScreenManager.java

```diff
@@ -0,0 +1,252 @@
+			LockScreenStatus status = getLockScreenStatus();
+			if (status == LockScreenStatus.REQUIRED) {
+				Intent showLockScreenIntent = new Intent(context.get(), SDLLockScreenActivity.class);
+				showLockScreenIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+				// Extra parameters for customization of the lock screen view
+				showLockScreenIntent.putExtra(SDLLockScreenActivity.LOCKSCREEN_ICON_EXTRA, lockScreenIcon);
+				showLockScreenIntent.putExtra(SDLLockScreenActivity.LOCKSCREEN_COLOR_EXTRA, lockScreenColor);
+				showLockScreenIntent.putExtra(SDLLockScreenActivity.LOCKSCREEN_CUSTOM_VIEW_EXTRA, customView);
+				showLockScreenIntent.putExtra(SDLLockScreenActivity.LOCKSCREEN_DEVICE_LOGO_EXTRA, deviceLogoEnabled);
+				showLockScreenIntent.putExtra(SDLLockScreenActivity.LOCKSCREEN_DEVICE_LOGO_BITMAP, deviceLogo);
+				context.get().startActivity(showLockScreenIntent);
+			} else if (status == LockScreenStatus.OFF) {
+				context.get().sendBroadcast(new Intent(SDLLockScreenActivity.CLOSE_LOCK_SCREEN_ACTION));
+			}
+		}
+	}
+
+	private boolean isForegrounded() {
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+			ActivityManager.RunningAppProcessInfo myProcess = new ActivityManager.RunningAppProcessInfo();
+			ActivityManager.getMyMemoryState(myProcess);
+			return myProcess.importance == ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND;
+		}
+		return true;
+	}
+
+	////
+	// HELPERS
+	////
+
+	/**
+	 * Step through some logic to determine if we need to show the lock screen or not
+	 * This function is usually triggered on some sort of notification.
+	 *
+	 * @return Whether or not the Lock Screen is required
+	 */
+	protected synchronized LockScreenStatus getLockScreenStatus() {
+
+		if ( (hmiLevel == null) || (hmiLevel.equals(HMILevel.HMI_NONE))) {
+			return LockScreenStatus.OFF;
+		}
+		else if ( hmiLevel.equals(HMILevel.HMI_BACKGROUND)) {
+			if (!driverDistStatus) {
+				//we don't have driver distraction, lock screen is entirely based on if user is using the app on the head unit
+				return LockScreenStatus.OFF;
+			} else {
+				return LockScreenStatus.REQUIRED;
+			}
+		}
+		else if ( (hmiLevel.equals(HMILevel.HMI_FULL)) || (hmiLevel.equals(HMILevel.HMI_LIMITED))) {
+			if (!driverDistStatus) {
+				return LockScreenStatus.OPTIONAL;
+			} else {
+				return LockScreenStatus.REQUIRED;
+			}
+		}
+		return LockScreenStatus.OFF;
+	}
+
+	private void downloadDeviceIcon(final String url){
+
+		if (deviceLogo != null || context.get() == null){
+			return;
+		}
+
+		new Thread(new Runnable(){
+			@Override
+			public void run(){
+				try{
+					deviceLogo = HttpUtils.downloadImage(url);
+					Intent intent = new Intent(SDLLockScreenActivity.LOCKSCREEN_DEVICE_LOGO_DOWNLOADED);
+					intent.putExtra(SDLLockScreenActivity.LOCKSCREEN_DEVICE_LOGO_EXTRA, deviceLogoEnabled);
+					intent.putExtra(SDLLockScreenActivity.LOCKSCREEN_DEVICE_LOGO_BITMAP, deviceLogo);
+					context.get().sendBroadcast(intent);
```

Comment:
Please add a null check for `context.get()` since this executes in a new thread a later time the previous check is not adequate.